integration_test.go

```diff
@@ -0,0 +1,231 @@
+package endless
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	IntegrationServerFeatureToggle = "GO_INTEGRATION_TEST_SERVER"
+	IntegrationServerPort          = "4242"
+	IntegrationServerPath          = "/test"
+)
+
+var (
+	serverpids = make(map[int]struct{})
+)
+
+// TestIntegrationServer isn't a real test - instead it is used as a
+// test server binary so that we can test signal handling and the subsequent
+// shutdown/restarts from a separate binary from the test itself.
+// This test is only run when an environment variable feature toggle
+// (named by IntegrationServerFeatureToggle) is set by the calling test
+func TestIntegrationServer(t *testing.T) {
+	if os.Getenv(IntegrationServerFeatureToggle) != "1" {
+		t.Skip("Skipping as do not need to stand up integration test server in this run")
```

Comment:
the language here is confusing, as this is never needed by a run, as described on the method comment.